drivers/ethernet/eth_mcux.c

```diff
@@ -1238,299 +1236,167 @@ static void eth_mcux_error_isr(const struct device *dev)
 }
 #endif
 
-#if DT_NODE_HAS_STATUS(DT_DRV_INST(0), okay)
-
-#if DT_INST_PROP(0, zephyr_random_mac_address) && \
-	NODE_HAS_VALID_MAC_ADDR(DT_DRV_INST(0))
-#error Conflict between 'local-mac-address' and 'zephyr,random-mac-address'
+#if defined(CONFIG_NOCACHE_MEMORY)
```

Comment:
Original code uses `#if defined(CONFIG_HAS_MCUX_CACHE)`. Why did we change?

Reply:
CONFIG_HAS_MCUX_CACHE tells that the SoC has cache, while CONFIG_NOCACHE_MEMORY means there is a non-cacheable region. The latter one is more fitting to the definition.